soh/soh/OTRGlobals.cpp

```diff
@@ -2041,11 +2041,14 @@ extern "C" int CustomMessage_RetrieveIfExists(PlayState* play) {
                ? CustomMessageManager::Instance->RetrieveMessage(Randomizer::hintMessageTableID, TEXT_ALTAR_ADULT)
                : CustomMessageManager::Instance->RetrieveMessage(Randomizer::hintMessageTableID, TEXT_ALTAR_CHILD);
         } else if (textId == TEXT_GANONDORF) {
-            if (INV_CONTENT(ITEM_ARROW_LIGHT) == ITEM_ARROW_LIGHT || !Randomizer_GetSettingValue(RSK_GANONDORF_LIGHT_ARROWS_HINT)) {
+            if (INV_CONTENT(ITEM_ARROW_LIGHT) == ITEM_ARROW_LIGHT || !Randomizer_GetSettingValue(RSK_LIGHT_ARROWS_HINT) ||
+                Randomizer_GetSettingValue(RSK_LIGHT_ARROWS_HINT) && Randomizer_GetSettingValue(RSK_GANONS_TRIALS)) {
```

Comment:
i don't think we need this change. the point of light arrow hint outside of trials is people need the lights to get past the trials. it would be out of logic to get to dorf without them, and i guess i see no reason to not give the light arrow hint to someone who decides to trial skip past the sheik hint without lights.